src/SAWScript/JavaTools.hs

```diff
@@ -0,0 +1,85 @@
+{- |
+Module      : SAWScript.JavaTools
+Description : Functionality for finding a Java executable and its properties.
+License     : BSD3
+Maintainer  : atomb
+Stability   : provisional
+-}
+
+module SAWScript.JavaTools
+  ( findJavaIn
+  , findJavaProperty
+  , findJavaMajorVersion
+  ) where
+
+import Control.Monad
+import Data.List.Extra (dropPrefix, firstJust, stripInfix)
+import Data.Maybe
+import System.Directory
+import System.Exit
+import System.Process
+
+-- | @'findJavaIn' javaBinDirs@ searches for an executable named @java@ in the
+-- directories in @javaBinDirs@. If @javaBinDirs@ is empty, then the @PATH@ is
+-- searched.
+--
+-- If the search finds at least one executable, then @Just@ the first result is
+-- returned. If no executables are found, then @Nothing@ is returned.
+findJavaIn :: [FilePath] -> IO (Maybe FilePath)
+findJavaIn javaBinDirs
+  | null javaBinDirs = findExecutable "java"
+  | otherwise        = listToMaybe <$> findExecutablesInDirectories javaBinDirs "java"
+
+-- | @'findJavaProperty' javaPath prop@ invokes @javaPath@ to dump its system
+-- properties (see <https://docs.oracle.com/javase/tutorial/essential/environment/sysprop.html>)
+-- and attempts to find the value associated with the property named @prop@.
+-- If such a value is found, then @Just@ that value is returned.
+-- If no property named @prop@ exists, then @Nothing@ is returned.
+--
+-- This will throw an exception if @javaPath@'s system properties cannot be
+-- determined.
+findJavaProperty :: FilePath -> String -> IO (Maybe String)
+findJavaProperty javaPath propertyNeedle = do
+  -- TODO RGS: Is it worth factoring this process-related code out? We do basically the same thing elsewhere in
```

Comment:
I'm going to move this code to a new `SAWScript.ProcessUtils` module, since I end up using this in a couple places here, and more places in #1046. (I considering putting it in `SAWScript.Utils`, but alas, that leads to import cycles.)